src/reactor.cc

```diff
@@ -0,0 +1,375 @@
+  } else {
+    throw cyclus::ValueError(
+        "Startup Failed: " + std::to_string(tritium_storage.quantity()) +
+        " kg in storage is less than required " +
+        std::to_string(startup_inventory) +
+        " kg to start-up!");
+  }
+}
+
+void Reactor::DecayInventory(
+    cyclus::toolkit::ResBuf<cyclus::Material>& inventory) {
+  if (!inventory.empty()) {
+    cyclus::Material::Ptr mat = inventory.Pop();
+    mat->Decay(context()->time());
+    inventory.Push(mat);
+  }
+}
+
+void Reactor::CombineInventory(
+    cyclus::toolkit::ResBuf<cyclus::Material>& inventory) {
+  if (!inventory.empty()) {
+    cyclus::Material::Ptr base = inventory.Pop();
+    int count = inventory.count();
+    for (int i = 0; i < count; i++) {
+      cyclus::Material::Ptr m = inventory.Pop();
+      base->Absorb(m);
+    }
+
+    inventory.Push(base);
+  }
+}
+
+void Reactor::ExtractHelium(
+    cyclus::toolkit::ResBuf<cyclus::Material>& inventory) {
+  if (!inventory.empty()) {
+    cyclus::Material::Ptr mat = inventory.Pop();
+    cyclus::CompMap c = mat->comp()->atom();
+    cyclus::compmath::Normalize(&c, mat->quantity());
+
+    // A threshold of 1e-5 was set to allow tritium_reserve inventories up to
+    // 1000kg. A 1 decade lower threshold prevents tritium_reserve inventories
+    // above 33kg.
+    cyclus::Material::Ptr helium = mat->ExtractComp(c[He3_id], He3_comp, 1e-5);
+
+    helium_storage.Push(helium);
+    inventory.Push(mat);
+  }
+}
+
+void Reactor::RecordEvent(std::string name, std::string val) {
+  context()
+      ->NewDatum("ReactorEvents")
+      ->AddVal("AgentId", id())
+      ->AddVal("Time", context()->time())
+      ->AddVal("Event", name)
+      ->AddVal("Value", val)
+      ->Record();
+}
+
+void Reactor::RecordOperationalInfo(std::string name, std::string val) {
+  context()
+      ->NewDatum("ReactorOperationsLog")
+      ->AddVal("AgentId", id())
+      ->AddVal("Time", context()->time())
+      ->AddVal("Event", name)
+      ->AddVal("Value", val)
+      ->Record();
+}
+
+void Reactor::RecordStatus(std::string status, double power) {
+  context()
+      ->NewDatum("ReactorStatus")
+      ->AddVal("AgentId", id())
+      ->AddVal("Time", context()->time())
+      ->AddVal("Status", status)
+      ->AddVal("Power", power)
+      ->Record();
+}
+
+void Reactor::RecordInventories(double storage, double excess, double sequestered, double blanket, double helium) {
+  context()
+      ->NewDatum("ReactorInventories")
+      ->AddVal("AgentId", id())
+      ->AddVal("Time", context()->time())
+      ->AddVal("TritiumStorage", storage)
+      ->AddVal("TritiumExcess", excess)
+      ->AddVal("TritiumSequestered", sequestered)
+      ->AddVal("LithiumBlanket", blanket)
+      ->AddVal("HeliumStorage", helium)
+      ->Record();
+}
+
+void Reactor::DepleteBlanket(double bred_tritium_mass) {
+  cyclus::Material::Ptr blanket_mat = blanket.Pop();
+
+  cyclus::CompMap b = blanket_mat->comp()->mass();
```

Comment:
If you get the `atom()` version, doesn't it make your life easier?

Reply:
Yes. Fixed.